src/iOS/Xamarin.iOS/Samples/Data/StatsQueryGroupAndSort/StatsQueryGroupAndSort.cs

```diff
@@ -748,40 +728,20 @@ public class ChooseStatisticOverlay : UIView
         public event EventHandler OnCanceled;
 
         // Constructor that takes a picker for defining new statistics.
-        public ChooseStatisticOverlay(CGRect frame, nfloat transparency, UIColor color, UIPickerView statPicker) : base(frame)
+        public ChooseStatisticOverlay(UIPickerView statPicker)
         {
-            // Store the statistics picker.
-            UIPickerView statisticPicker = statPicker;
-
-            // Create a semi-transparent overlay with the specified background color.
-            BackgroundColor = color;
-            Alpha = transparency;
-
-            // Set the total height and width of the control set.
-            nfloat totalHeight = 400;
-            nfloat totalWidth = 320;
-
-            // Find the bottom x and y of the view.
-            nfloat centerX = Frame.Width / 2;
-            nfloat centerY = Frame.Bottom - 40;
-
-            // Find the start x and y for the control layout (aligned to the bottom of the view).
-            nfloat controlX = centerX - totalWidth / 2;
-            nfloat controlY = centerY - totalHeight;
-
+            this.TranslatesAutoresizingMaskIntoConstraints = false;
             // Toolbar with "Add" and "Done" buttons.
-            UIToolbar toolbar = new UIToolbar
-            {
-                BarStyle = UIBarStyle.Black,
-                Translucent = false
-            };
-            toolbar.SizeToFit();
+            UIToolbar toolbar = new UIToolbar();
+            toolbar.TranslatesAutoresizingMaskIntoConstraints = false;
+
+            statPicker.TranslatesAutoresizingMaskIntoConstraints = false;
 
             // Add Button (add the new stat and don't dismiss the UI).
-            UIBarButtonItem addButton = new UIBarButtonItem("Add", UIBarButtonItemStyle.Done, (s, e) =>
+            UIBarButtonItem addButton = new UIBarButtonItem("Add", UIBarButtonItemStyle.Plain, (s, e) =>
```

Comment:
Almost forgot one more:
Clicking "Add" button on "Stat definitions" screen does nothing unless user changes something in the picker first.